doc/source/ray-air/examples/batch_tuning.ipynb

```diff
@@ -0,0 +1,1801 @@
+    "\n",
+    "    # Train model.\n",
+    "    model = model.fit(train_X, train_y)\n",
+    "    pred_y = model.predict(valid_X)\n",
+    "\n",
+    "    # Evaluate.\n",
+    "    error = sklearn.metrics.mean_absolute_error(valid_y, pred_y)\n",
+    "\n",
+    "    # Save a model checkpoint.\n",
+    "    # https://docs.ray.io/en/latest/tune/tutorials/tune-checkpoints.html\n",
+    "    checkpoint = ray.air.checkpoint.Checkpoint.from_dict(\n",
+    "        {\"model\": model, \"location_id\": location_id}\n",
+    "    )\n",
+    "\n",
+    "    # Save checkpoint and report back metrics, using ray.air.session.report()\n",
+    "    # The metrics you specify here will appear in Tune summary table.\n",
+    "    # They will also be recorded in Tune results under `metrics`.\n",
+    "    metrics = dict(error=error)\n",
+    "    session.report(metrics, checkpoint=checkpoint)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "38a2d25f",
+   "metadata": {},
+   "source": [
+    "## Run batch training on Ray Tune <a class=\"anchor\" id=\"run_tune_search\"></a>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "68f5d7ee",
+   "metadata": {},
+   "source": [
+    "**In the cell below, we configure the resources allocated per trial.** \n",
+    "\n",
+    "Tune uses this resources allocation to control the parallelism. For example, if each trial was configured to use 4 CPUs, and the cluster had only 32 CPUs, then Tune will limit the number of concurrent trials to 8 to avoid overloading the cluster. For more information, see [A Guide To Parallelism and Resources](https://docs.ray.io/en/master/tune/tutorials/tune-resources.html#tune-parallelism)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "id": "015c314b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# 3. Customize resources per trial, here we set 1 CPU each.\n",
+    "train_model = tune.with_resources(train_model, {\"cpu\": 1})"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ac8b4f58",
+   "metadata": {},
+   "source": [
+    "### Kick off a Ray Tune experiment!\n",
+    "\n",
+    "Recall what we are doing, high level, is training several different models per dropoff location.  We are using Ray Tune so we can run all these trials in parallel.  At the end, we will inspect the results of the experiment and deploy only the best model per dropoff location.\n",
+    "\n",
+    "**In the cell below, we use AIR configs and run the experiment using `tuner.fit()`.**\n",
+    "\n",
+    "Tune will report on experiment status, and after the experiment finishes, you can inspect the results.\n",
+    "\n",
+    "Notice in the AIR config, we have specified a local directory `my_Tune_logs` for logging instead of the default `~/ray_results` directory. Learn more about logging Tune results at [How to configure logging in Tune](https://docs.ray.io/en/master/tune/tutorials/tune-output.html#tune-logging).\n",
```

Comment:
why do we do this btw?

Reply:
Giving your logs a project name makes them easier to find. Also giving a relative path, means you can see your logs inside the Jupyter browser. 

I'll update this in the notebook.